src/js/media-chrome-listbox.js

```diff
@@ -0,0 +1,278 @@
+
+    return composedPath[index];
+  }
+
+  handleSelection(e) {
+    const item = this.#getItem(e);
+
+    if (!item) return;
+
+    const selected = item.getAttribute('aria-selected') === 'true';
+
+    if (this.getAttribute('aria-multiselectable') !== 'true') {
+      this.#assignedElements.forEach(el => el.setAttribute('aria-selected', 'false'));
+    }
+
+    if (selected) {
+      item.setAttribute('aria-selected', 'false');
+    } else {
+      item.setAttribute('aria-selected', 'true');
+    }
+  }
+
+  handleMovement(e) {
+    const { key } = e;
+    const els = this.#items;
+
+    let currentOption = this.#getItem(e);
+    if (!currentOption) {
+      currentOption = els.filter(el => el.getAttribute('tabindex') === '0')[0];
+    }
+
+    let nextOption;
+    
+    switch (key) {
+      case 'ArrowDown':
+        nextOption = currentOption.nextElementSibling;
+
+        if (nextOption.hasAttribute('disabled')) {
+          nextOption = nextOption.nextElementSibling;
+        }
+
+        break;
+      case 'ArrowUp':
+        nextOption = currentOption.previousElementSibling;
+
+        if (nextOption.hasAttribute('disabled')) {
+          nextOption = nextOption.previousElementSibling;
+        }
+
+        break;
+      case 'Home':
+        nextOption = els[0];
+        break;
+      case 'End':
+        nextOption = els[els.length - 1];
+        break;
+      default:
+        nextOption = this.#searchItem(key);
+        break;
+    }
+
+    if (nextOption) {
+      els.forEach(el => el.setAttribute('tabindex', '-1'));
+      nextOption.setAttribute('tabindex', '0');
+      nextOption.focus();
+    }
+  }
+
+  handleClick(e) {
+    const item = this.#getItem(e);
+
+    if (!item || item.hasAttribute('disabled')) return;
+
+    this.#items.forEach(el => el.setAttribute('tabindex', '-1'));
+    item.setAttribute('tabindex', '0');
+
+    this.handleSelection(e);
+  }
+
+  #searchItem(key) {
+    this.#clearKeysOnDelay();
+
+    const els = this.#items;
+    const activeIndex = els.findIndex(el => el.getAttribute('tabindex') === '0');
+    const repeatedKey = this.#keysSoFar === key || this.#keysSoFar.length === 0;
+
+    // don't accumulate keys if it's a repeated key
+    // unless we haven't started accumulating a key yet
+    if (!repeatedKey || this.#keysSoFar.length === 0) {
+      this.#keysSoFar += key;
+    }
+
+    // if it's a repeat key, skip the current item
+    const after = els.slice(activeIndex + (repeatedKey ? 1 : 0)).filter(el => el.textContent.startsWith(this.#keysSoFar));
+    const before = els.slice(0, activeIndex - 1).filter(el => el.textContent.startsWith(this.#keysSoFar));
+
+    return [...after, ...before][0];
+  }
+
+  #clearKeysOnDelay() {
+    window.clearTimeout(this.#clearKeysTimeout);
+    this.#clearKeysTimeout = null;
+
+    this.#clearKeysTimeout = window.setTimeout(() => {
+      this.#keysSoFar = '';
+      this.#clearKeysTimeout = null;
+    }, 500);
```

Comment:
idea(non-blocking): Should we make this an attr/prop with 500ms the default to help w/a11y for folks who need more time typing?

Reply:
yeah, I can see the benefit of having it.

Reply:
Given that the w3c example implementation uses 500, I'd say we leave it as is until we need it.